### PR TITLE
Added Azure.Redis.PublicNetworkAccess rule and refactored Redis YAML rules.

### LINKED ISSUE #934: Connect to Azure Cache for Redis instance using private endpoints
# Rule request

## Suggested rule change

Use private endpoints to connect to Redis instances.

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.Cache/Redis**

## Additional context

[Azure Cache for Redis with Azure Private Link](https://docs.microsoft.com/azure/azure-cache-for-redis/cache-private-link)

### LINKED ISSUE #935: Limit public access for Azure Cache for Redis instances
# Rule request

## Suggested rule change

Limit public network access when private endpoints are configured by ensuring `publicNetworkAccess` is `Disabled`.

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.Cache/Redis**

## Additional context

[Azure Cache for Redis with Azure Private Link](https://docs.microsoft.com/azure/azure-cache-for-redis/cache-private-link)

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fixes #935.

Added `Azure.Redis.PublicNetworkAccess` rule.

Also refactored some rules into YAML. #1109

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [x] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Nice work! Let build the module docs though. Otherwise all good to merge.

**Comment by ArmaanMcleod** [comment]
> @ArmaanMcleod Nice work! Let build the module docs though. Otherwise all good to merge.

@BernieWhite Good catch, some reason I completely forgot to build docs this time 😄. Might need to get onto creating a github workflow to do that automatically.

**Comment by ArmaanMcleod** [comment]
@BernieWhite Docs updated. Seems like the license/cla status is hanging.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod All good. Thanks Armaan.

File: docs/CHANGELOG-v1.md [hunk 1]
```diff
@@ -14,6 +14,11 @@ See [troubleshooting guide] for a workaround to this issue.
 
 ## Unreleased
 
+What's changed since pre-release v1.13.0-B2202063:
+
+- New rules:
+  - Limit public access for Azure Cache for Redis instances. [#935](https://github.com/Azure/PSRule.Rules.Azure/issues/935)
```
Comment (BernieWhite) [inline docs/CHANGELOG-v1.md:20]:
```suggestion
- New rules:
  - Azure Cache for Redis:
    - Limit public access for Azure Cache for Redis instances. [#935](https://github.com/Azure/PSRule.Rules.Azure/issues/935)
```

**Comment by BernieWhite** [comment]
@ArmaanMcleod Bump a commit and it should resolve it. It's probably just a temporary issue.

**Comment by BernieWhite** [comment]
@ArmaanMcleod Looks like it is a bigger problem. Try later tonight.

**Pull request merged**